auth-foundation/src/test/java/com/okta/authfoundation/client/NetworkUtilsTest.kt

```diff
@@ -354,6 +364,371 @@ class NetworkingTest {
         }
         assertThat((result as OidcClientResult.Success<String>).result).isEqualTo("bar")
     }
+
+    @Test fun testPerformRequestWithNon429ErrorHasNoRetry(): Unit = runTest {
+        oktaRule.enqueue(path("/test")) { response ->
+            response.setResponseCode(401)
+            response.setBody("""{"foo":"bar"}""")
+        }
+        oktaRule.enqueue(path("/test")) { response ->
+            throw(IllegalStateException("Retried request after getting non-429 response."))
```

Comment:
This is a super nit pick, but the syntax here is a little different than I'd expect.
```suggestion
            throw IllegalStateException("Retried request after getting non-429 response.")
```

Similar thing in other tests as well.

You can also remove the `response ->` from the line above in these.

Reply:
Sure, that sounds good. Removing ```response ->``` gives me an override ambiguity error for some reason, so I'm leaving that in

Reply:
You can make response `_` instead, which will remove the warning.